### DISCUSSION TITLE
Version 3 Status?

### DISCUSSION THREAD

**Discussion by GaryKopp** [griffon/griffon#321] [General]
I'd like to try Griffon, but I need JavaFX 17 support. I see that is coming in version 3, which seems to be in "pre-release." How risky would it be for me to use the github development_3_X branch at this point?
--Gary Kopp

**Comment by aalmiray** [comment]
Hi Gary,
Griffon 3.x is still ongoing. The core functionality is almost complete and plugins have been updated as far as I recall. There's low risk of APIs changing in a dramatic way at this point (we renamed packages to remove split packages for example) but we're not yet ready to go fully JPMS compatible which would be a boon for JavaFx apps. 

The risk today is that you'd have to build everything from the ground up (core and plugins) and continue to do so until we post a release. 

I must confess I'm spending more time these days working with JReleaser than Griffon. I hope I can come back to it before the end of the year.

**Comment by GaryKopp** [comment]
Thanks, Andres. Perhaps I'll use Griffon 2.16 instead. Then I can either use JDK 8 with JavaFX (and even GroovyFX, I guess) _or_ I can use Swing instead of JavaFX. In the case of Swing I have a question I haven't been able to find an answer to. JDK 8 is the minimum requirement for Griffon 2.16, but what is the maximum? Will Griffon 2.16 with Swing work with JDK 17? Thanks again. P.S.: is Griffon 2.16 compatible with Groovy 3.X?

**Reply by aalmiray** [reply]
As far as I can tell Java 17+ and Swing should be OK.

**Reply by GaryKopp** [reply]
I'm not sure that's correct. In your Gitter feed (which I just discovered) on 17 Dec 2021 tor.sh reported that not only did JavaFX 17 not work with Griffon 2.16 (which we know is true), but the **Swing** archetype doesn't work with JDK 17 either. Your reply then only addressed JavaFX, so things seem to be a bit uncertain :-(

**Reply by aalmiray** [reply]
Griffon archetypes work, but it's the use of reflection by the framework that breaks in Java 17+ dues to the IllegalAcces flag that wa added to 11 was turned off in 17. See

Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @33bb9311
	at com.google.inject.internal.cglib.core.$ReflectUtils$1.run(ReflectUtils.java:61)
	at com.google.inject.internal.cglib.core.$ReflectUtils.<clinit>(ReflectUtils.java:52)
	at com.google.inject.internal.cglib.reflect.$FastClassEmitter.<init>(FastClassEmitter.java:67)

**Comment by GaryKopp** [comment]
Thanks. Let me explore one more thing before deciding on my approach. Would you prefer that I _not_ try Griffon 3 with JavaFX 17? If I were to try v3 I might have many questions since the doc is obviously incomplete. I also might identify issues/bugs that I would have to report and hopefully get some answers. I'm thinking you don't have time for any of that at this point! I'm guessing there really isn't anyone else actively working on the project.
